theano/sandbox/cuda/opt.py

```diff
@@ -2380,21 +2380,26 @@ def local_gpu_extract_diagonal(node):
     gpu_from_host(extract_diagonal) -> extract_diagonal(gpu_from_host)
 
     """
-    if (isinstance(node.op, nlinalg.ExtractDiag) and
-        isinstance(node.inputs[0].type,
-                   theano.tensor.TensorType)):
+    if ((isinstance(node.op, nlinalg.ExtractDiag) or
+         isinstance(node.op, theano.tensor.Diagonal)) and
+        isinstance(node.inputs[0].type, theano.tensor.TensorType)):
         inp = node.inputs[0]
+        if (isinstance(node.op, theano.tensor.Diagonal) and
+            (node.op.offset, node.op.axis1, node.op.axis2) != (0, 0, 1)):
```

Comment:
Having such integers free in the wild is not very good. Since there is multiple places where we test if a diagonal op has default properties, I suggest we add a method Diagonal.has_default_properties() and add each default values as static attributes.  

```python
class Diagonal(Op):
    __props__ = ("offset", "axis1", "axis2")
    default_offset = 0
    default_axis1 = 0
    default_axis2 = 0

    def __init__(self, offset=default_offset, axis1=default_axis1, axis2=default_axis2):
        ...
```